lib/ansible/modules/cloud/amazon/ec2_vpc_net.py

```diff
@@ -136,198 +136,206 @@
+                except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+                    module.fail_json_aws(e, "Unable to disassociate {0}. You must detach or delete all gateways and resources that "
+                                         "are associated with the CIDR block before you can disassociate it.".format(association_id))
 
         if dhcp_id is not None:
             try:
                 if update_dhcp_opts(connection, module, vpc_obj, dhcp_id):
                     changed = True
-            except BotoServerError as e:
-                module.fail_json(msg=e)
+            except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+                module.fail_json_aws(e, "Failed to update DHCP options")
 
         if tags is not None or name is not None:
             try:
-                if update_vpc_tags(connection, module, vpc_obj, tags, name):
+                if update_vpc_tags(connection, module, vpc_id, tags, name):
                     changed = True
-            except BotoServerError as e:
-                module.fail_json(msg=e)
+            except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+                module.fail_json_aws(e, msg="Failed to update tags")
 
-        # Note: Boto currently doesn't currently provide an interface to ec2-describe-vpc-attribute
-        # which is needed in order to detect the current status of DNS options. For now we just update
-        # the attribute each time and is not used as a changed-factor.
-        try:
+        current_dns_enabled = connection.describe_vpc_attribute(Attribute='enableDnsSupport', VpcId=vpc_id)['EnableDnsSupport']['Value']
+        current_dns_hostnames = connection.describe_vpc_attribute(Attribute='enableDnsHostnames', VpcId=vpc_id)['EnableDnsHostnames']['Value']
+        if current_dns_enabled != dns_support:
+            changed = True
             if not module.check_mode:
-                connection.modify_vpc_attribute(vpc_obj.id, enable_dns_support=dns_support)
-                connection.modify_vpc_attribute(vpc_obj.id, enable_dns_hostnames=dns_hostnames)
-        except BotoServerError as e:
-            e_msg=boto_exception(e)
-            module.fail_json(msg=e_msg)
-
-        if not module.check_mode:
-            # get the vpc obj again in case it has changed
-            try:
-                vpc_obj = connection.get_all_vpcs(vpc_obj.id)[0]
-            except BotoServerError as e:
-                e_msg=boto_exception(e)
-                module.fail_json(msg=e_msg)
+                try:
+                    connection.modify_vpc_attribute(VpcId=vpc_id, EnableDnsSupport={'Value': dns_support})
+                except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+                    module.fail_json_aws(e, "Failed to update enabled dns support attribute")
```

Comment:
Does the second parameter need `msg=` or does it just work ok without?

Reply:
Tested to be sure, and it works without.